File: backend-laravel/app/Models/Permiso.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

/**
 * Catálogo FIJO de permisos del sistema (tabla `permisos`), no editable
 * por la institución. `plataforma` (movil/escritorio) es el techo de lo
 * que ese permiso puede llegar a habilitar, no una asignación — ver
 * narrativa, "Capa de Permisos (fija, definida por el sistema)".
 */
class Permiso extends Model
{
    protected $table = 'permisos';

    protected $primaryKey = 'id_permiso';

    protected $fillable = [
        'nombre',
        'plataforma',
        'descripcion',
    ];

    public function roles(): BelongsToMany
    {
        return $this->belongsToMany(
            Rol::class,
            'roles_permisos',
            'permiso_id',
            'rol_id',
            'id_permiso',
            'id_rol'
        );
    }
}

File: app_flutter/lib/models/permiso.dart
/// Espejo de lo que devuelve `PermisosController` (backend-laravel) — un
/// permiso del catálogo FIJO del sistema (tabla `permisos`). No lo edita
/// la institución: solo se usa para armar el checklist de
/// `RolesScreen`. `plataforma` es el techo de lo que ese permiso puede
/// llegar a habilitar ('movil' o 'escritorio'), ver `App\Models\Permiso`
/// en el backend.
class Permiso {
  const Permiso({
    required this.id,
    required this.nombre,
    required this.plataforma,
    required this.descripcion,
  });

  factory Permiso.fromJson(Map<String, dynamic> json) {
    return Permiso(
      id: json['id_permiso'] as int,
      nombre: json['nombre'] as String,
      plataforma: json['plataforma'] as String,
      descripcion: json['descripcion'] as String?,
    );
  }

  final int id;
  final String nombre;

  /// 'movil' | 'escritorio'.
  final String plataforma;
  final String? descripcion;
}
